pages/secure/sec_graylog_beats.rst

```diff
@@ -0,0 +1,102 @@
+	[ -f /usr/lib/jvm/jre/lib/security/cacerts ] && cp /usr/lib/jvm/jre/lib/security/cacerts /etc/graylog/server/cacerts.jks
+	[ -f /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/security/cacerts ] && cp /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/security/cacerts /etc/graylog/server/cacerts.jks
+	 
+	 
+	# import CA .der into keystore
+	# will only work if the default password & user is not changed.
+	keytool -importcert -alias shadowCA -keystore /etc/graylog/server/cacerts.jks -storepass changeit -file shadowCA.der
+
+custom JVM Keystore for Graylog
+-------------------------------
+
+Modify the :ref:`JVM Setting <default_file_location>` to include ``-Djavax.net.ssl.trustStore=/etc/graylog/server/cacerts.jks`` in the ``GRAYLOG_JAVA_OPTS``.
+
+create certificates
+===================
+
+Create certificates for each server, all hostnames and IPs that might be used later to connect from and to this server should be included in the certificates. See `README of shadowCA <https://github.com/graylog-labs/shadowCA#create-certificates>`__ for the possible options. The most common error is that the certificate name does not match the hostname that is used for the connection.
+
+The shadowCA uses the same settings that can be found :ref:`in the SSL <ssl_setup>` documentation, but easy up the process. 
+
+
+Deploy and configure
+====================
+
+Graylog
+-------
+
+HTTPS
+^^^^^
+
+Place the ``.key`` and ``.crt file`` on your Graylog server in the configuration dir (/etc/graylog/server/) and add them to the Graylog server.conf. In addition change the ``rest_listen_uri`` and ``web_listen_uri`` to **https**. You might need to cover other settings in a multinode cluster or special setups - just read the comments of the settings inside of the server.conf.
+
+When using the collector-sidecar, use the **https** URI in the ``sidecar_configuration.yml``.
+
+After restart of Graylog the web interface and the API is served via https only. No automatic redirect from http to https is made.
+
+TLS Beats Input
+^^^^^^^^^^^^^^^
+
+To enable TLS on the input, a certificate (and private key file) is needed. It can be the same or a different certificate as the one of your REST/web interface, as long as it matches all hostnames of your input. Just reference the files `TLS cert file` and `TLS private key file` in the Beats Input configuration and restart the input. 
+
+The ingesting client will verify the presented certificate against his know CA certificates, if that is successful communication will be establised using TLS. 
+
+
+Add client authentication to beats input
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Create one directory (``/etc/graylog/server/trusted_clients``) that will hold all client certificates you allow to connect to the beats input - that must be available on all Graylog server that have the input enabled. Write that path in the beats input configuration `TLS Client Auth Trusted Certs` and select **required** for the option `TLS client authentication`.
```

Comment:
Just a suggestion for easier reading. Split the first sentence in two:

`Create one directory (``/etc/graylog/server/trusted_clients``) that will hold all client certificates you allow to connect to the beats input. This directory must be available on all Graylog server that have the input enabled.`